modules/javafx.controls/src/test/java/test/javafx/scene/control/ControlPropertiesTest.java

```diff
@@ -0,0 +1,225 @@
+import com.sun.javafx.scene.control.InputField;
+import com.sun.javafx.scene.control.IntegerField;
+import com.sun.javafx.scene.control.WebColorField;
+import com.sun.javafx.scene.control.skin.FXVK;
+
+/**
+ * Tests contract for properties and their accessors and mutators
+ * in the Control type hierarchy.
+ *
+ * Currently uses a list of classes, so any new Controls must be added manually.
+ * Perhaps the test should scan modulepath and find all the Controls automagically.
+ */
+public class ControlPropertiesTest {
+
+    private static final boolean FAIL_FAST = true;
+
+    // list all current descendants of Control class.
+    private Set<Class> allControlClasses() {
+        return Set.of(
+            Accordion.class,
+            ButtonBar.class,
+            ButtonBase.class,
+            Button.class,
+            Cell.class,
+            CheckBox.class,
+            CheckBoxListCell.class,
+            CheckBoxTableCell.class,
+            CheckBoxTreeCell.class,
+            CheckBoxTreeTableCell.class,
+            ChoiceBox.class,
+            ChoiceBoxTreeCell.class,
+            ChoiceBoxTreeTableCell.class,
+            ColorPicker.class,
+            ComboBox.class,
+            ComboBoxBase.class,
+            ComboBoxTreeCell.class,
+            ComboBoxTreeTableCell.class,
+            DateCell.class,
+            DatePicker.class,
+            DoubleField.class,
+            FXVK.class,
+            //HTMLEditor.class,
+            Hyperlink.class,
+            IndexedCell.class,
+            InputField.class,
+            IntegerField.class,
+            Labeled.class,
+            Label.class,
+            ListCell.class,
+            ListView.class,
+            MenuBar.class,
+            MenuButton.class,
+            Pagination.class,
+            PasswordField.class,
+            ProgressIndicator.class,
+            ProgressBarTreeTableCell.class,
+            RadioButton.class,
+            ScrollBar.class,
+            ScrollPane.class,
+            Separator.class,
+            SeparatorMenuItem.class,
+            Slider.class,
+            Spinner.class,
+            SplitPane.class,
+            TableRow.class,
+            TableView.class,
+            TabPane.class,
+            TextArea.class,
+            TextField.class,
+            TextFieldTreeCell.class,
+            TextFieldTreeTableCell.class,
+            TitledPane.class,
+            ToggleButton.class,
+            TreeCell.class,
+            TreeTableCell.class,
+            TreeTableRow.class,
+            TreeTableView.class,
+            WebColorField.class
+        );
+    }
+
+    /**
+     * Tests for missing final keyword in Control properties and their accessors/mutators.
+     */
+    @Test
+    public void testMissingFinalMethods() {
+        for (Class c : allControlClasses()) {
+            check(c);
+        }
+    }
+
+    private void check(Class cls) {
+        Method[] methods = cls.getMethods();
```

Comment:
This will only return public methods. Since you modified `checkModifiers` to check protected methods, you will need to use `getDeclaredMethods` instead. This will return protected (and package scope) methods as well as public methods. It won't return any method in a superclass, but since the set of classes you are checking includes intermediate classes (such as `ButtonBase` and `Labeled`), you will still get the coverage (and without redundantly checking the same method in those classes multiple times).

Reply:
Right, what was I thinking??
And we also have Region.setWidth() which is protected and not final by design.  Reverting the last change.